auth/models.py

```diff
@@ -93,7 +93,7 @@ def create_or_get_user(provider_id, name, username, email, avatar_url):
         document = {
             'id': id_,
             'provider_id': provider_id,
-            'username': username,
+            'username': username.lower(),
```

Comment:
@anuveyatsu Why are we lowercasing here, when saving?  
`get_user_by_username` lowercases before check anyway after your refactor here, right?
```
ret = session.query(User).filter(func.lower(User.username) == func.lower(username_)).first()
```

Think we should save username as is so that actual links have real usernames init. Eg https://datahub.io/Zelima is actual url but it will work if https://datahub.io/zelima typed and vice versa,  wdyt?

Reply:
I believe it is a common practice to lowercase usernames, e.g., slack, instagram, facebook, twitter etc.

We're lowercasing in `get_user_by_username` function since we already have usernames with uppercases.

Reply:
@anuveyatsu Agree, also this avoids to have multiple users  with the same username, but case differentiated